### PR TITLE
Add `PublisherHandler` unit test

### LINKED ISSUE #418: Unit test for `PublisherHandler`
**Is your feature request related to a problem? Please describe.**
Add a unit test for the `PublisherHandler`

**Describe the solution you'd like**
- The `PublisherHandler` unit test is tested.
- `NatsConn` will probably need to be mocked
- Don't use mockgen because it's archived

**Describe alternatives you've considered**
- testify
- mockery
- gomockery
- mockgen
**Additional context**

### PR DESCRIPTION
**Pull request by chr12c**
Closes #418 

Add `PublisherHandler` unit test and mock the nats connection using `testify`

## Tasks

- [x] Add unit test for `PublisherHandler`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -84,6 +85,9 @@ test-argo:
 test-triggers:
 	$(MAKE) -C triggers/run-completion-event-trigger test functional-test
 
+test-provider-service:
+	$(MAKE) -C provider-service test
+
 test-all: test helm-test-operator helm-test-provider test-argo test-triggers
```
Comment (TobyPinfold) [inline Makefile:91]:
```suggestion
test-all: test helm-test-operator helm-test-provider test-argo test-triggers test-provider-service
```

**Review by aidandunlop — APPROVED** [review]

**Review by TobyPinfold — APPROVED** [review]

File: triggers/run-completion-event-trigger/internal/publisher/publisher_handler.go [hunk 2]
```diff
@@ -27,16 +27,20 @@ func (e *ConnectionError) Error() string {
 	return e.Message
 }
 
+type NatsConn interface {
```
Comment (grahamia) [inline triggers/run-completion-event-trigger/internal/publisher/publisher_handler.go:30]: This seems like a generic interface for publishing but is restricted by its name to NatsConn. Maybe just Publisher?

File: triggers/run-completion-event-trigger/internal/publisher/publisher_handler.go
Comment (grahamia) [inline triggers/run-completion-event-trigger/internal/publisher/publisher_handler.go:1]: As this is just limited to Nats at the moment and not generic and New is explicitly returning a Nats implementation I think it should be called nats_publisher_handler.go or something

File: triggers/run-completion-event-trigger/internal/publisher/publisher_handler.go (hunk 2 above)
Comment (chr12c) [reply triggers/run-completion-event-trigger/internal/publisher/publisher_handler.go:30]: This interface is required in order to mock the behaviour of the nats conn. The real NatsConn isn't an interface so I'm relying on golang's duck typing to get enable the mocking.
`NatsPublisher.Publish` calls `NatsConn.Publish`.

File: triggers/run-completion-event-trigger/internal/publisher/publisher_handler.go
Comment (chr12c) [reply triggers/run-completion-event-trigger/internal/publisher/publisher_handler.go:1]: renaming it back to `NewNatsPublisher`
https://github.com/sky-uk/kfp-operator/pull/419/commits/5235d4e57572834b8bdf681de6bc3dd247e7f2ea

File: triggers/run-completion-event-trigger/internal/publisher/publisher_handler.go (hunk 2 above)
Comment (chr12c) [reply triggers/run-completion-event-trigger/internal/publisher/publisher_handler.go]: https://github.com/sky-uk/kfp-operator/pull/419/commits/5235d4e57572834b8bdf681de6bc3dd247e7f2ea
I've made it lowercase for the moment so nothing outside can implement it

Comment (chr12c) [reply triggers/run-completion-event-trigger/internal/publisher/publisher_handler.go]: I think if we do end up with multiple implementations of the `PublisherHandler`, they will split into different packages, with `natsConn` moving into the nats-only package.

**Review by grahamia — APPROVED** [review]

**Pull request merged**